hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMPrepareRequest.java

```diff
@@ -185,7 +185,6 @@ private static long waitForLogIndex(long minOMDBFlushIndex,
     // If we purge logs without waiting for this index, it may not make it to
     // the RocksDB snapshot, and then the log entry is lost on this OM.
     long minRatisStateMachineIndex = minOMDBFlushIndex + 1; // for the ratis-metadata transaction
```

Comment:
@nandakumar131 , just found that We should also remove `minRatisStateMachineIndex` and just use `minOMDBFlushIndex`.

Reply:
Do we really need the `lastRatisAppliedIndex` check?
Can't we just rely on `om.getRatisSnapshotIndex()` and make sure that the given `minOMDBFlushIndex` is present in the snapshot? If `minOMDBFlushIndex` is present in the snapshot, then it will definitely be applied in the Ratis' state machine.

Reply:
Thanks for the explanation. Updated the patch accordingly.